src/LocalTranslationsBatcher.php

```diff
@@ -113,18 +112,32 @@ public function createBatch(array $langcodes = []): void {
     }
 
     $extensions = $this->getExtensionsToTranslate();
-    if (!$extensions) {
+    if (!$extensions || !$langcodes) {
       return;
     }
 
-    locale_translation_flush_projects();
-    locale_translation_check_projects_local($extensions, $langcodes);
-    $options = _locale_translation_default_update_options();
-    $batch = locale_translation_batch_fetch_build($extensions, $langcodes, $options);
+    // Build operations for local translation batch import.
+    $operations = [];
+    foreach ($extensions as $extension) {
+      $operations[] = [
+        [get_class($this), 'importProjectPoFiles'],
+        [$extension, $langcodes],
+      ];
+    }
+
+    $batch = [
+      'operations' => $operations,
+      'title' => t('Updating translations.'),
+      'progress_message' => '',
+      'error_message' => t('Error importing translation files'),
+    ];
+
     batch_set($batch);
-    if ($batch = locale_config_batch_update_components($options, $langcodes)) {
+    // Update config translations.
+    if ($batch = locale_config_batch_update_components([], $langcodes)) {
```

Comment:
Let's also add an output message before starting with configuration import and one after like:
```
Importing configuration translations...
Done.
```